.github/workflows/check-pull-request.yaml

```diff
@@ -1,33 +1,76 @@
-on:
-  pull_request:
-    branches:
-      - main
-  repository_dispatch:
-    types: trigger-check-pull-request
+on: [push, pull_request]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
+      # The type of runner that the job will run on
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-    # Checks-out your repository under $GITHUB_WORKSPACE
-    - name: Checkout
-      uses: actions/checkout@v2
-
-    # Install dependencies
-    - name: Install Dependencies
-      run: |
-        pip install --upgrade pip
-        pip install -r requirements.txt
-
-    # Make test
-    - name: Make test
-      run: |
-        make test
+
+      #----------------------------------------------
+      #       check-out repo and set-up python     
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      
+      #----------------------------------------------
+      #          install & configure poetry         
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.1.6
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true  
+
+      # Install dependencies
+      - name: pip Install Dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -r requirements.txt
+
+      #----------------------------------------------
+      #       load cached venv if cache exists      
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+
+      #----------------------------------------------
+      # install dependencies if cache does not exist 
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+        
+      #----------------------------------------------
+      #    install your root project, if required 
+      #----------------------------------------------      
+      - name: Install library
+        run: poetry install --no-interaction
+          
+      - name: Run image
+        uses: abatilo/actions-poetry@v2.0.0
+        # with:
+        #   poetry-version: ${{ matrix.poetry-version }}
+      - name: View poetry --help
+        run: poetry --help
```

Comment:
Is there any specific reason for having to log available poetry commands in every run? I think we can remove this step, so we can reduce the amount of content being logged by the Action run.

Reply:
yeah, good catch. this was something I did while learning.